src/tensorwaves/optimizer/callbacks.py

```diff
@@ -30,6 +31,50 @@ def finalize(self) -> None:
         self.__progress_bar.close()
 
 
+class YAMLSummary:
+    def __init__(
+        self,
+        filename: str,
+        step_size: int = 10,
+    ) -> None:
+        """Log fit parameters and the estimator value to a `tf.summary`.
+
+        The logs can be viewed with `TensorBoard
+        <https://www.tensorflow.org/tensorboard>`_ via:
+
+        .. code-block:: bash
+
+            tensorboard --logdir logs
+        """
+        self.__iteration = 0
+        self.__step_size = step_size
+        self.__stream = open(filename, "w")
+
+    def __call__(self, parameters: dict, estimator_value: float) -> None:
```

Comment:
The Callbacks are actually called on every function evaluation... Unfortunately we don't have access to the iterations (at least when using minuit).

Reply:
Yeah, that was one of the reasons for posting #165

Reply:
I meant more that `iterations` should be renamed to `function_calls`. As iteration is the incorrect word here